apps/open_dashboard_backend/backend.cc

```diff
@@ -0,0 +1,331 @@
+
+		if (!mUdpReceiver.DataAvailable())
+			break;
+		const size_t len = mUdpReceiver.Receive(packet.GetData(), packet.GetBufferSize());
+
+		for (int i=0; i<packet.GetNumberOfMessages(); ++i)
+		{
+			MessageType messageType = static_cast<MessageType>(packet.GetNextMessageType());
+
+			switch (messageType)
+			{
+			case CONTROL_MSG:
+			{
+				control = packet.GetNextMessage<Control>();
+				mControlDataDynamicModel.SetState(static_cast<ControlDataDynamicModel::State>(control.state()));
+				break;
+			}
+			case VEHICLE_DYNAMIC_MSG:
+			{
+				vehicleDynamic = packet.GetNextMessage<VehicleDynamic>();
+				mVehicleDataModel.SetVehicleVelocity(vehicleDynamic.velocity());
+				mVehicleDataModel.SetVehicleAx(vehicleDynamic.ax());
+				mVehicleDataModel.SetVehicleAy(vehicleDynamic.ay());
+				mVehicleDataModel.SetVehicleAz(vehicleDynamic.az());
+				break;
+			}
+			case GPS_MSG:
+			{
+				gps = packet.GetNextMessage<Gps>();
+				mVehicleDataModel.SetPositionLon(gps.lon());
+				mVehicleDataModel.SetPositionLat(gps.lat());
+
+				mControlDataDynamicModel.SetLat(gps.lat());
+				mControlDataDynamicModel.SetLon(gps.lon());
+				break;
+			}
+			case POWERTRAIN_MSG:
+			{
+				powertrain = packet.GetNextMessage<Powertrain>();
+
+				mVehicleDataModel.SetEngineRpm(powertrain.rotation());
+				mVehicleDataModel.SetEnginePowerOut(powertrain.engine_power_out());
+				mVehicleDataModel.SetEngineTorque(powertrain.engine_torque());
+				mVehicleDataModel.SetEngineFuelConsumptionAbs(powertrain.engine_fuel_consumption_abs());
+				mVehicleDataModel.SetEngineFuelConsumptionAct(powertrain.engine_fuel_consumption_act());
+				mVehicleDataModel.SetEngineFuelConsumptionAvg(powertrain.engine_fuel_consumption_avg());
+				mVehicleDataModel.SetEngineFuelLevel(powertrain.engine_fuel_level());
+
+				mVehicleDataModel.SetGearboxGear(powertrain.gear());
+				mVehicleDataModel.SetGearboxPowerIn(powertrain.gearbox_power_in());
+				mVehicleDataModel.SetGearboxPowerOut(powertrain.gearbox_power_out());
+				mVehicleDataModel.SetGearboxRotationIn(powertrain.gearbox_rotation_in());
+				mVehicleDataModel.SetGearboxRotationOut(powertrain.gearbox_rotation_out());
+				break;
+			}
+			case DRIVER_INPUT_MSG:
+			{
+				driverInput = packet.GetNextMessage<DriverInput>();
+				mVehicleDataModel.SetInputThrottle(driverInput.throttle());
+				mVehicleDataModel.SetInputBrake(driverInput.brake());
+				mVehicleDataModel.SetInputSteeringWheelAngle(driverInput.steering_wheel_angle());
+				break;
+			}
+			default:
+				break;
+			}
+
+			std::stringstream serializedMessages;
+			serializedMessages << MessageToString(vehicleDynamic);
+			serializedMessages << MessageToString(gps);
+			serializedMessages << MessageToString(powertrain);
+			serializedMessages << MessageToString(driverInput);
+
+			auto& serializedQuantities = mControlDataDynamicModel.GetSerializedQuantities();
+			std::string quantityString;
+			for (int j = 0; std::getline(serializedMessages, quantityString, '\n'); ++j)
```

Comment:
this is hard to follow, what are you trying to do here? can we make this a method?